docs/04-reference/wingsdk-spec.md

```diff
@@ -253,105 +253,115 @@ resource Bucket {
   /**
    * Run an inflight whenever an object is uploaded to the bucket.
    */
-  onCreate(fn: inflight (key: str) => void, opts: BucketOnUploadProps?): void;
+  onCreate(fn: IBucketEventHandler, opts?: BucketOnUploadProps): void;
```

Comment:
I think the main reason reason we have `IBucketEventHandler` is because TypeScript doesn't have a way to represent inflight function types. I think we should probably avoid leaking this implementation detail in Wing if possible, to me it makes more sense to keep the existing type. The existing signature is also easier to read since it doesn't require looking up what `IBucketEventHandler` means. does that make sense?

Also curious if @eladb has any thoughts

Reply:
I reverted the change for now while we wait for @eladb opinion.

Reply:
@eladb Can you take a look?

Reply:
I agree with @Chriscbr 💅